test/test_prototype_transforms_consistency.py

```diff
@@ -480,16 +501,22 @@ def test_signature_consistency(config):
         )
 
     extra = prototype_params.keys() - legacy_params.keys()
-    extra_without_default = {param for param in extra if prototype_params[param].default is not inspect.Parameter.empty}
+    extra_without_default = {
+        param
+        for param in extra
+        if prototype_params[param].default is inspect.Parameter.empty
+        and prototype_params[param].kind not in {inspect.Parameter.VAR_POSITIONAL, inspect.Parameter.VAR_KEYWORD}
+    }
     if extra_without_default:
         raise AssertionError(
-            f"The prototype transform requires the parameters {sequence_to_str(sorted(missing), separate_last='and ')}, "
-            f"but the legacy transform does not. Please add a default value."
+            f"The prototype transform requires the parameters "
+            f"{sequence_to_str(sorted(extra_without_default), separate_last='and ')}, but the legacy transform does "
+            f"not. Please add a default value."
         )
 
-    for name, legacy_param in legacy_params.items():
-        prototype_param = prototype_params[name]
-        assert prototype_param.kind is legacy_param.kind
+    legacy_kinds = {name: param.kind for name, param in legacy_params.items()}
```

Comment:
Instead of going through the parameters one by one and fail on the first mismatch, we now compute everything upfront. This has the benefit that all errors are visible and we don't have to re-run the test.